### PR TITLE
Update optimizer: allow custom optimizer

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot/pull/2128
requires https://github.com/Drakkar-Software/OctoBot-Trading/pull/787

### CODE REVIEW

File: octobot/community/graphql_requests.py [hunk 1]
```diff
@@ -40,6 +40,10 @@
         _id
         profitability
     }
+    stats {
+        _id
+        profitability
+    }
```
Comment (Herklos) [inline octobot/community/graphql_requests.py:46]: ?

Comment (GuillaumeDSM) [reply octobot/community/graphql_requests.py:46]: it's from the backtester branch, I'm removing it

File: octobot/strategy_optimizer/optimizer_settings.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        # return [
+        #     {
+        #         optimizer_constraint.OptimizerConstraint.NAME_KEY: "tentacle-user_input",
+        #         optimizer_constraint.OptimizerConstraint.MIN_VAL_KEY: 1,
+        #         optimizer_constraint.OptimizerConstraint.MAX_VAL_KEY: 3,
+        #         optimizer_constraint.OptimizerConstraint.MIN_STEP_KEY: 0.25,
+        #         optimizer_constraint.OptimizerConstraint.MAX_STEP_KEY: 1,
+        #         optimizer_constraint.OptimizerConstraint.STAY_WITHIN_BOUNDARIES_KEY: False,
+        #     }
+        # ]
```
Comment (Herklos) [inline octobot/strategy_optimizer/optimizer_settings.py:169]: What should we do with this code ?

**Review by Herklos — APPROVED** [review]
Great job on the optimiser 💯

File: octobot/strategy_optimizer/optimizer_settings.py (hunk 2 above)
Comment (GuillaumeDSM) [reply octobot/strategy_optimizer/optimizer_settings.py:169]: it's an example of a constraint declaration

**Review by Herklos — APPROVED** [review]

**Pull request merged**